sys/include/evtimer_msg_mbox.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2020 Simon Brummer <simon.brummer@posteo.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  sys_evtimer
+ * @{
+ *
+ * @file
+ * @brief       IPC Message based evtimer event, extended with mbox support.
+ *
+ * @author      Simon Brummer <simon.brummer@posteo.de>
+ */
+#ifndef EVTIMER_MSG_MBOX_H
+#define EVTIMER_MSG_MBOX_H
+
+#include "assert.h"
+#include "msg.h"
+#include "mbox.h"
+#include "evtimer.h"
+#include "evtimer_msg.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   IPC-message event with optional mbox support.
+ * @extends evtimer_msg_event_t
+ */
+typedef struct {
+    evtimer_event_t event;   /**< event base class */
+    msg_t msg;               /**< IPC message so send on event */
+    mbox_t *target_mbox;     /**< mbox the IPC message can be stored */
+} evtimer_msg_mbox_event_t;
+
+/**
+ * @brief   Initialze evtimer event.
+ *
+ * @param[in,out] event  Event to initialize.
+ */
+static inline void evtimer_msg_mbox_event_init(evtimer_msg_mbox_event_t *event)
+{
+    assert(event);
+    event->target_mbox = NULL;
+}
```

Comment:
This is what I meant above
```suggestion
static inline void evtimer_msg_mbox_event_init(evtimer_msg_mbox_event_t *event,
                                        mbox_t *target_mbox)
{
    assert(event);
    event->target_mbox = target_mbox;
}
```

Reply:
I thought on removing the init function entirely. evtimer_msg forces the users to setup the contents of the msg within the struct. I'll assume this is done for performance reasons. With that in mind, offering a init function and setting the message content outside of the init function seems to be odd. 

I see two options there:
a) Remove the init function and relay that users set target_mbox themself.
b) Set the mbox on evtimer_add_mbox via a parameter.

I tend to implement option "a" as it keeps the initialization process consistent. Do you have any thoughts on that topic?